xls_parsing/history.py

```diff
@@ -0,0 +1,51 @@
+from common import locate_header
+
+class History:
+    def __init__(self, name, bunk, past_activities, past_preferences):
+        self.name = name
+        self.bunk = bunk
+        self.past_activities = past_activities
+        self.past_preferences = past_preferences
+
+def parse_sheet(sheet):
+    """Takes in an Excel sheet and spits out objects containing camper histories"""
+    header_row = sheet[1]
+
+    create_history = _parse_header(header_row)
+    # Create list of history objects to return
+    histories = [create_history(row) for row in sheet.iter_rows(min_row=2)]
+
+    return histories
+
+
+def _parse_header(header_row):
+    """Takes in the row we believe to be the header, returns a function that creates history objects"""
+    header_name = "Chanich Name"
+    header_bunk = "Chanich Tzrif"
+    # Determine number of past activities -- assumes same number of preference columns as activity columns
+    counter = 0
```

Comment:
Yeah this is totally a fine way to count the number of activities. If you wanted to do it a fancy way, you could do either something like
```python
len([cell for rcell in header_row if cell.value])
```
or
```python
sum(1 for cell in header_row if "Activity" in cell.value)
```
Either way I'm 99% sure you want `cell.value` and not `header_row[cell].value`; `cell` is the cell not the cell's location.

Reply:
I think that's actually your problem:
```
TypeError: tuple indices must be integers or slices, not Cell
```
is saying you can't index into something with a `Cell`, which is what you're doing when you say `header_row[cell]`.